### PR TITLE
Add pow and sqrt to stdlib

### PR DESCRIPTION
**Pull request by raoulvdberge**
This pull request adds pow and sqrt.

### CODE REVIEW

**Comment by hikari-no-yume** [comment]
Hmm, you'd need to forbid negative powers.

Also, this makes me wonder about what other integer math functions are missing.

**Comment by raoulvdberge** [comment]
Fixed in latest commit.

Missing functions: Sin, cos, tan, sqrt, min, max, log, log10, random (use Math.random for this?), PI value

File: src/value-types.js [hunk 1]
```diff
@@ -81,6 +81,12 @@ function IntegerValue(value) {
         /* remainder with the sign of the divisor */
         return IntegerValue(((value % b) + b) % b);
     };
+    that.pow = function(b) {
+        if (b.getValue() < 0) {
+            throw new Error(value + ' is negative');
```
Comment (hikari-no-yume) [inline src/value-types.js]: Surely you mean `b.getValue() + ' is negative'`? Otherwise, that would be a _very_ confusing error message ^^

**Comment by raoulvdberge** [comment]
Woops!

**Comment by raoulvdberge** [comment]
Pow and sqrt go together so I've also added that.

File: src/stdlib.js [hunk 2]
```diff
@@ -132,6 +132,12 @@ defunIntBinop('div', function (a, b) {
 defunIntBinop('mod', function (a, b) {
     return a.mod(b);
 });
+defunIntBinop('pow', function (a, b) {
+    return a.pow(b);
+});
+defunTyped('sqrt', ['integer'], ['integer'], function (stack, scope) {
+    stack.push(stack.pop().sqrt());
```
Comment (hikari-no-yume) [inline src/stdlib.js:139]: Why aren't you using `defunIntBinop` here?

Comment (raoulvdberge) [inline src/stdlib.js:139]: `defunIntBinop` requires 2 arguments, while `sqrt` only takes one.

Comment (hikari-no-yume) [inline src/stdlib.js:139]: Oh, right. My bad.

**Comment by raoulvdberge** [comment]
Can this be merged? My objects pull request kind of depends on this :P.

**Comment by hikari-no-yume** [comment]
Hmm. Are you sure it always produces an integer result?

**Comment by hikari-no-yume** [comment]
...it doesn't. That's not good. Actually, I'm not sure square root is going to be useful often for an integer, given you can't have a fractional part... though again, I'm not sure, maybe there's a use for it. If we are to add sqrt, you'd also need cube root probably.

On a different note, you may want to rebase this branch to squash all the changes into one commit, so I don't have to.

**Comment by raoulvdberge** [comment]
Well, I just added it _so it is there_ until a solution for #8 is found.

**Comment by raoulvdberge** [comment]
I'll close this for now until there is support for floating point numbers.

**Pull request closed without merging**